specification/verifiedid/Microsoft.VerifiedId/main.tsp

```diff
@@ -0,0 +1,79 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Azure.ResourceManager;
+
+@armProviderNamespace("Microsoft.VerifiedId")
+@service({
+  title: "VerifiedIdMgmtClient",
+})
+@versioned(Microsoft.VerifiedId.Versions)
+@doc("VerifiedId Resource Provider management API.")
+@armCommonTypesVersion(Azure.ResourceManager.CommonTypes.Versions.v5)
+namespace Microsoft.VerifiedId;
+
+/**
+ * Supported versions for Microsoft.VerifiedId.
+ */
+enum Versions {
+  /**
+   * The 2024-01-26-preview version of the Microsoft.VerifiedId resource provider.
+   */
+  @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
+  v2024_01_26_preview: "2024-01-26-preview",
+}
+
+interface Operations extends Azure.ResourceManager.Operations {}
+
+@doc("A VerifiedId authority resource")
+model Authority is TrackedResource<AuthorityProperties> {
+  @doc("The ID of the authority")
+  @pattern("^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$")
+  @key("name")
```

Comment:
>name [](http://example.com/codeflow?start=8&length=4)

Recommend something more specific than "name", e.g. authorityName

Reply:
This change results in an error saying name is a required property:
Resource types must include a string property called 'name'.TypeSpec(@azure-tools/typespec-azure-resource-manager/arm-resource-missing-name-property)

Should I override this error in some way or follow the check from azure-tools?

Reply:
Oh just the @key change is needed, can do